merf/merf.py

```diff
@@ -45,104 +51,98 @@ def __init__(
         gll_early_stop_threshold=None,
         max_iterations=20,
     ):
+        # Note fixed_effects_model must already be instantiated when passed in.
+        self.fixed_effects_model = fixed_effects_model
         self.gll_early_stop_threshold = gll_early_stop_threshold
         self.max_iterations = max_iterations
 
-        self.cluster_counts = None
-        # Note fixed_effects_model must already be instantiated when passed in.
-        self.fe_model = fixed_effects_model
-        self.trained_fe_model = None
-        self.trained_b = None
-
-        self.b_hat_history = []
-        self.sigma2_hat_history = []
-        self.D_hat_history = []
-        self.gll_history = []
-        self.val_loss_history = []
-
-    def predict(self, X: np.ndarray, Z: np.ndarray, clusters: pd.Series):
+    def predict(self, X: ArrayLike):
         """
         Predict using trained MERF.  For known clusters the trained random effect correction is applied.
         For unknown clusters the pure fixed effect (RF) estimate is used.
 
         Args:
-            X (np.ndarray): fixed effect covariates
-            Z (np.ndarray): random effect covariates
-            clusters (pd.Series): cluster assignments for samples
+            X: predictors (both fixed and random effect covariates)
```

Comment:
That conflicts with the sklearn interface spec: https://scikit-learn.org/stable/developers/develop.html#instantiation. I think it's quite standard that the `X` input to predict method should be the same structure as for the fit, otherwise any ML algorithm will fail. I will add some more documentation anyway.